src/SfxWeb/src/app/modules/powershell-commands/command/command.component.ts

```diff
@@ -42,12 +43,20 @@ export class CommandComponent implements OnInit{
     return this.inputForm.controls['optionalInputs'] as UntypedFormGroup;
   }
 
-  updateInvalidInputText() {
+  updateInputs() {
     let invalids = [];
     for (let name in this.requiredInputs?.controls) {
-      if (!this.requiredInputs.controls[name]?.valid)
+      this.command.getParam(name).value = this.requiredInputs.controls[name].value;
+      if (!this.requiredInputs.controls[name]?.valid){
         invalids.push(name);
+      }
     }
     this.invalidInputs = invalids.join(', ');
+
+    for (let name in this.optionalInputs?.controls) {
```

Comment:
when is optionalinputs null?

Reply:
they were erroring out on page load as inputForm was not set yet, undefined/null was being returned by the getters. Checking again today that doesn't seem to be the case anymore(???). Removed.